workers/generic-worker/artifacts.go

```diff
@@ -360,23 +359,7 @@ func (task *TaskRun) uploadArtifact(artifact artifacts.TaskArtifact) *CommandExe
 }
 
 func copyToTempFileAsTaskUser(filePath string) (tempFilePath string, err error) {
-	// We want to run generic-worker, which is os.Args[0] if we are running generic-worker, but if
-	// we are running tests, os.Args[0] will be the test executable, so then we use relative path to
-	// installed binary. This hack will go if we can impersonate the logged on user.
-	var exe string
-	if testing.Testing() {
-		exe = os.Getenv("GOPATH")
-		switch runtime.GOOS {
-		case "windows":
-			exe += `\bin\generic-worker.exe`
-		default:
-			exe += "/bin/generic-worker"
-		}
-	} else {
-		exe = os.Args[0]
-	}
```

Comment:
Pulled this out into a helper to reduce duplicative code and this also makes it so some other functions don't need to pass the `exe` string through anymore.

https://github.com/taskcluster/taskcluster/pull/6679/files#diff-47bb9ac1b0971421de4be385184e1f8ba471bf0cc53654f0ee5e3b033457bb5b